pkg/scaling/resolver/azure_keyvault_handler.go

```diff
@@ -0,0 +1,77 @@
+/*
```

Comment:
I'm not a Go expert but let's make sure we surface throttling errors from Azure Key Vault since it has low service limits.

Ideally we would also:
- Cache secret for a few minutes
- Re-fetch cached secret if it was unauthorized to see if it was rotated or not.

Reply:
Currently doing the documentation changes, will start on this right after.